projects/storefrontlib/cms-components/myaccount/payment-methods/payment-methods.component.ts

```diff
@@ -96,24 +96,28 @@ export class PaymentMethodsComponent implements OnInit {
   }
 
   deletePaymentMethod(paymentMethod: PaymentDetails): void {
-    this.userPaymentService.deletePaymentMethod(paymentMethod.id);
-    this.editCard = null;
+    if (paymentMethod.id) {
+      this.userPaymentService.deletePaymentMethod(paymentMethod.id);
+      this.editCard = undefined;
```

Comment:
Looks like `this.editCard = undefined;` should be outside of the `if`

Reply:
only payment method is deleted, then this `editCard` can be undefined.